tracing-appender/src/non_blocking.rs

```diff
@@ -258,7 +263,14 @@ impl Drop for WorkerGuard {
             .sender
             .send_timeout(Msg::Shutdown, Duration::from_millis(100))
         {
-            Ok(_) | Err(SendTimeoutError::Disconnected(_)) => (),
+            Ok(_) => {
+                /* Attempt to wait for `Worker` to flush all messages before dropping. This happens
+                when the `Worker` calls `recv()` on a zero-capacity channel. Use `send_timeout`
+                so that drop is not blocked indefinitely.
+                */
```

Comment:
nit: elsewhere, we use `// ...` for comments, even when they span multiple lines --- i think this would be the first `/* ... */` block comment in the repo if it merges...